File: Progressio_backend/Progressio.Services/Database/Entities/Episode.cs
namespace Progressio.Services.Database.Entities;

public class Episode
{
    public int Id { get; set; }
    public int SeasonId { get; set; }
    public int EpisodeNumber { get; set; }
    public string Title { get; set; } = null!;
    public int? DurationMinutes { get; set; }
    public DateTime AirDate { get; set; }

    public Season Season { get; set; } = null!;
    public ICollection<EpisodeProgress> EpisodeProgresses { get; set; } = [];
    public ICollection<ContentComment> Comments { get; set; } = [];
    public ICollection<CharacterVote> CharacterVotes { get; set; } = [];
}

File: ui/progressio_desktop/lib/model/episode.dart
class Episode {
  final int id;
  final int seasonId;
  final int episodeNumber;
  final String title;
  final int? durationMinutes;
  final DateTime? airDate;
  final String? description;

  Episode({
    this.id = 0,
    this.seasonId = 0,
    this.episodeNumber = 1,
    this.title = '',
    this.durationMinutes,
    this.airDate,
    this.description,
  });

  factory Episode.fromJson(Map<String, dynamic> json) {
    return Episode(
      id: json['id'] ?? 0,
      seasonId: json['seasonId'] ?? 0,
      episodeNumber: json['episodeNumber'] ?? 1,
      title: json['title'] ?? '',
      durationMinutes: json['durationMinutes'],
      airDate: json['airDate'] != null ? DateTime.parse(json['airDate']) : null,
      description: json['description'],
    );
  }
}